umm-spec-lib/src/cmr/umm_spec/validation/data_date.clj

```diff
@@ -0,0 +1,50 @@
+(ns cmr.umm-spec.validation.data-date
+  "Defines validations for UMM collection DataDates and MetadataDates"
+  (:require
+   [clj-time.core :as time]
+   [clojure.string :as string]
+   [cmr.umm-spec.date-util :as date-util]))
+
+(defn data-dates-warning-validation
+  "Validate that DataDates/MetadataDates' 
+   CREATE, UPDATE, REVIEW and DELETE date values are in the right order and range: 
+   CREATE needs to be <= UPDATE and they both need to be in the past. 
+   REVIEW needs to be <= DELETE and they both need to be in the future.
+   field-path is :DataDates/:MetadataDates, 
+   value is the value of :DataDates/:MetadataDates."
+  [field-path value]
+  (let [latest-create (date-util/latest-date-of-type value "CREATE")
+        earliest-update (date-util/earliest-date-of-type value "UPDATE")
+        latest-update (date-util/latest-date-of-type value "UPDATE")
+        earliest-review (date-util/earliest-date-of-type value "REVIEW")
+        latest-review (date-util/latest-date-of-type value "REVIEW")
+        earliest-delete (date-util/earliest-date-of-type value "DELETE")]
+    (def warning-msgs (atom ""))
```

Comment:
I don't think you need to use an atom here. I think what you want to do is return a list of errors, not the errors all as one string. 

So it should be something like `(let [errors (concat (when .... [error message]))` then do a `(when (seq errors))` at the end.